shared/src/main/scala/amf/client/remod/AMFClient.scala

```diff
@@ -11,29 +11,30 @@ import amf.core.resolution.pipelines.ResolutionPipeline
 import amf.core.services.RuntimeCompiler
 import amf.core.validation.AMFValidationReport
 
-import scala.concurrent.Future
-
+import scala.concurrent.{ExecutionContext, Future}
 
 // client
-private[remod] class AMFClient(env: BaseEnvironment) {
+private[amf] class AMFClient(val configuration: AMFConfiguration) {
 
+  implicit val exec: ExecutionContext = configuration.getExecutionContext
 
-  def getEnvironment:BaseEnvironment = env
+  def getEnvironment: AMFConfiguration = configuration
```

Comment:
If `configuration` is a public `val` then there is no need for this method. I would probably make the val private.